numpy/core/src/multiarray/datetime.c

```diff
@@ -2846,6 +2846,16 @@ convert_pyobject_to_timedelta(PyArray_DatetimeMetaData *meta, PyObject *obj,
         return 0;
     }
     else {
+        if (PyArray_IsScalar(obj, Integer)) {
```

Comment:
Can't this be done as `else if` with the error return alone in an `else` section?